pkg/controller/userdatasecret/userdatasecret_controller.go

```diff
@@ -0,0 +1,217 @@
+package userdatasecret
+
+import (
+	"context"
+
+	wkl "github.com/openshift/windows-machine-config-operator/pkg/controller/wellknownlocations"
+	wmc "github.com/openshift/windows-machine-config-operator/pkg/controller/windowsmachine"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh"
+	corev1 "k8s.io/api/core/v1"
+	k8sapierrors "k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kubeTypes "k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	ControllerName = "controller_userdatasecret"
+	userDataSecret = "windows-user-data"
+)
+
+var log = logf.Log.WithName(ControllerName)
+
+/**
+* USER ACTION REQUIRED: This is a scaffold file intended for the user to modify with their own Controller
+* business logic.  Delete these comments after modifying this file.*
+ */
+
+// Add creates a new UserDataSecret Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager, clusterServiceCIDR string) error {
+	reconciler, err := newReconciler(mgr)
+	if err != nil {
+		return errors.Wrapf(err, "could not create %s reconciler", ControllerName)
+	}
+	return add(mgr, reconciler)
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) (reconcile.Reconciler, error) {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := client.New(cfg, client.Options{Scheme: mgr.GetScheme()})
+	if err != nil {
+		return nil, err
+	}
+
+	signer, err := wmc.CreateSigner()
+	if err != nil {
+		return nil, errors.Wrapf(err, "error creating signer using private key: %v", wkl.PrivateKeyPath)
+	}
+	reconciler := &ReconcileUserDataSecret{client: client, scheme: mgr.GetScheme(), signer: signer}
+
+	// Create userData secret
+	userDataSecret, _ := reconciler.createUserDataSecret()
+	err = reconciler.client.Create(context.TODO(), userDataSecret)
```

Comment:
aren't we creating this as a part of reconcile?

Reply:
The primary resource for this controller is `Secret`. The reconcile will be triggered only for `Update` and `Delete` events. The other alternative for this could be creating `UserData` in `windowsmachine-controller`